src/bots/bot.ts

```diff
@@ -25,33 +26,57 @@ export class Bot implements IBot {
   /**
    * Create a Bot instance.
    *
-   * @param username - The username or key for the account
-   * @param password - The password or secret for the account
-   * @param urls - A non empty array of HTTP URLs as a list of strings
-   * @param rateLimit - The rate-limit(per minute) under which this bot must be used
+   * @param config - Of type {@link BotConfig}. browserConfig takes type {@link BrowserConfig}
    * @returns An instance of the Bot class
    *
    * * @example
    * ```ts
-   * const bot = new Bot('username', 'password', 100)
+   * const bot = new Bot({
+   *  credentials: { // optional
+   *    username: 'username',
+   *    password: 'password'
+   *  },
+   *  urls: ['www.example.com'],
+   *  rateLimit: 100, // optional, per minute
+   *  browserConfig: {
+   *    executablePath: '/path/to/your/Chromium',
+   *    headless: false,
+   *    useStealthPlugin: true,
+   *    useAdblockerPlugin: true,
+   *    blockDomains: [
+   *      'some-tracker.io',
+   *      'image-host.net',
+   *    ],
+   *    blockResourceTypes: ['image', 'media', 'stylesheet', 'font'],
+   *    proxy: {
+   *      address: '99.99.99.99',
+   *      port: 9999,
+   *      credentials: {
+   *        username: 'proxyuser1',
+   *        password: 'proxypass1',
+   *      },
+   *    }
+   *  }
+   * })
    * ```
    *
    * @beta
    */
   // eslint-disable-next-line max-params
-  constructor (username: string, password: string, urls: string[], rateLimit?: number) {
-    this.username = username
-    this.password = password
-    if(urls.length === 0) {
-      throw new Error('urls cannot be an empty array as the Bot will never be selected otherwise')
+  constructor (botConfig: BotConfig) {
+    if(typeof botConfig.credentials !== 'undefined' && botConfig.urls.length === 0) {
+      throw new Error('Bots with credentials cannot have urls as the Bot will never be selected otherwise')
```

Comment:
```suggestion
      throw new Error('Bots with credentials must have urls as the Bot will never be selected otherwise')
```

It currently says every bot who has credentials must not have urls - which is not what it wants to say I think

Reply:
Ah, you're right. It should `cannot have zero urls`. I'll make it `must have at least one url`